libheif/compression.h

```diff
@@ -17,17 +17,28 @@
  * You should have received a copy of the GNU Lesser General Public License
  * along with libheif.  If not, see <http://www.gnu.org/licenses/>.
  */
-#ifndef LIBHEIF_METADATA_COMPRESSION_H
-#define LIBHEIF_METADATA_COMPRESSION_H
+#ifndef LIBHEIF_COMPRESSION_H
+#define LIBHEIF_COMPRESSION_H
 
 #include <vector>
 #include <cinttypes>
 #include <cstddef>
 
-#if WITH_DEFLATE_HEADER_COMPRESSION
+#include <error.h>
+
+#if WITH_ZLIB_COMPRESSION
 std::vector<uint8_t> deflate(const uint8_t* input, size_t size);
 
-std::vector<uint8_t> inflate(const std::vector<uint8_t>&);
+Error inflate_zlib(const std::vector<uint8_t>&compressed_input, std::vector<uint8_t> *output);
+
+Error inflate_deflate(const std::vector<uint8_t>&, std::vector<uint8_t> *output);
```

Comment:
Funny name. What does this do?

Reply:
zlib can be run in "zlib" (RFC 1951) or "deflate" (RFC 1950) header modes. There doesn't appear to be a way to say "just deal with both" on decode (i.e. when inflating compressed data).

So we had deflate() and inflate(). I extended the API to handle the inflate() options for both deflate compression and zlib compression, based on the configured compression from the `cmpC` box. That got put into the name - its the inflate process for the deflate compression mode.

Possibly inflate_defl(), to match the 4CC code? The downside to that is that the compression / decompression code is really independent from the 23001-17 Amd 2 process. inflate_rfc1950() but that is a bit obscure too.

Reply:
Yes, I would also prefer `compress_deflate`, `decompress_deflate`, `decompress_brotli` ...

Reply:
I'll do the renaming after the merge.

Reply:
Thanks Dirk.